src/lib/types/minions.ts

```diff
@@ -101,6 +103,18 @@ export interface AlchingActivityTaskOptions extends ActivityTaskOptions {
 	alchValue: number;
 }
 
+export interface FarmingActivityTaskOptions extends ActivityTaskOptions {
+	plantsName: string;
+	channelID: string;
+	quantity: number;
+	upgradeType: 'compost' | 'supercompost' | 'ultracompost' | '';
+	patchType: PatchTypes.PatchData;
+	getPatchType: string;
+	msg: KlasaMessage;
```

Comment:
no no no! very bad, it seem like you pass an entire Message just to access the author, when the user is already accessible in the task. Regardless this is a bad idea, as we want to pass only atomic, simple data structures, not a complex class like a Message.